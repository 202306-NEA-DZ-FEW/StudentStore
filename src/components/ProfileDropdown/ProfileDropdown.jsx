import Image from "next/image";
import Link from "next/link";
import { useRouter } from "next/router";
import { useState } from "react";
import { toast } from "react-toastify";

import { useAuth } from "@/context/AuthContext";

function ProfileDropdown() {
    const { currentUser, logout } = useAuth();
    const route = useRouter();
    const [isDropdownOpen, setDropdownOpen] = useState(false);

    const toggleDropdown = () => {
        setDropdownOpen(!isDropdownOpen);
    };
    // logout function
    async function handleLogout() {
        try {
            await logout();
            route.push("/signin");
        } catch (error) {
            toast.error("Failed to log out");
        }
    }
    return (
        <>
            {currentUser ? (
                <div className='dropdown dropdown-end'>
                    <label
                        tabIndex={0}
                        onClick={toggleDropdown}
                        className='btn btn-ghost hover:bg-slate-200 btn-circle avatar'
                    >
                        <div className='w-8 rounded-full'>
                            <Image
                                src={
                                    currentUser.photoURL ||
                                    "/images/profile.jpg"
                                }
                                width={20}
                                height={20}
                                alt='profile-pic'
                            />
                        </div>
                    </label>
                    {isDropdownOpen && (
                        <ul
                            tabIndex={0}
                            className='mt-1 z-[1] px-1 py-2 shadow menu menu-sm dropdown-content  text-black bg-[#F1F6FA] rounded-md w-22'
                        >
                            <li>
                                <Link
                                    href='#'
                                    className='text-[#7874F2] font-semibold justify-between hover:text-grey'
                                >
                                    Profile
                                </Link>
                            </li>
                            <li>
                                <Link
                                    href='#'
                                    className='text-[#7874F2] font-semibold  hover:text-grey'
                                >
                                    Listings
                                </Link>
                            </li>
                            <li>
                                <button
                                    className='text-[#7874F2] font-semibold   hover:text-grey'
                                    onClick={handleLogout}
                                >
                                    Logout
                                </button>
                            </li>
                        </ul>
                    )}
                </div>
            ) : (
                <Link
                    className='bg-[#585785] py-1 px-1 rounded-md text-white hover:bg-[#6d6d9a] transition duration-300'
                    href='/signin'
                >
                    Sign In
                </Link>
            )}
        </>
    );
}

export default ProfileDropdown;
