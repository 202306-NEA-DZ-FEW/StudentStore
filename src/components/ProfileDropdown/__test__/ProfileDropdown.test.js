import renderer from "react-test-renderer";

import ProfileDropdown from "../ProfileDropdown";

it("renders correctly", () => {
    const tree = renderer.create(<ProfileDropdown />).toJSON();
    expect(tree).toMatchSnapshot();
});
