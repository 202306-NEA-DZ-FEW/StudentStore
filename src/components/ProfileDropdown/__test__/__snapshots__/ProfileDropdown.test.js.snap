// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="dropdown dropdown-end"
>
  <label
    className="btn btn-ghost btn-circle avatar"
    onClick={[Function]}
    tabIndex={0}
  >
    <div
      className="w-10 rounded-full"
    >
      <img
        alt="profile-pic"
        data-nimg="1"
        decoding="async"
        height={20}
        onError={[Function]}
        onLoad={[Function]}
        src=""
        style={
          Object {
            "color": "transparent",
          }
        }
        width={20}
      />
    </div>
  </label>
</div>
`;
